src/Module/Notifications/Notification.php

```diff
@@ -74,31 +89,36 @@ public static function rawContent(array $parameters = [])
 	}
 
 	/**
+	 * {@inheritDoc}
+	 *
 	 * Redirect to the notifications main page or to the url for the chosen notifications
 	 *
-	 * @return string|void
+	 * @throws HTTPException\NotFoundException In case the notification is either not existing or is not for this user
 	 * @throws HTTPException\InternalServerErrorException
+	 * @throws \Exception
 	 */
 	public static function content(array $parameters = [])
 	{
+		if (!local_user()) {
+			notice(DI::l10n()->t('Permission denied.'));
+			return Login::form();
+		}
+
 		$request_id = $parameters['id'] ?? false;
 
 		if ($request_id) {
-			try {
-				$notify = DI::notify()->getByID($request_id);
-				DI::notify()->setSeen(true, $notify);
+			$notify = DI::notify()->getByID($request_id);
```

Comment:
This needs to verify the `local_user()` is the owner of the notification and issue a `Forbidden` error if they aren't.

Reply:
In https://github.com/friendica/friendica/blob/23c64b9a119c5df9f985ad4c62ef6fb79d68bee3/src/BaseRepository.php#L73 we throw a `NotFoundException` 
and at 
https://github.com/friendica/friendica/blob/23c64b9a119c5df9f985ad4c62ef6fb79d68bee3/src/Repository/Notify.php#L71
we already use the `local_user()` as a second parameter.

So because of this situation, we throw all the time a `NotFoudException` in both cases we're not authorized or it doesn't exist. I'd like to be more "general" to avoid guessing of notification IDs.

And so the `NotFoundException` is already captured at the general Module class, so I don't see a reason to re-caputre it. Therefore I removed the complete try-catch :-)

Reply:
It's a good idea. However I would avoid to use `local_user()` in the Notify repository and would rather have 
```php
public function getByID(int $id, int $uid)
{
	return $this->selectFirst(['id' => $id, 'uid' => $uid]);
}
```
and in the module that cares about the authentication:
```php
$notify = DI::notify()->getByID($request_id, local_user());
```
What do you think?